virtool/mongo/connect.py

```diff
@@ -32,9 +35,26 @@ async def connect(db_connection_string: str, db_name: str) -> AsyncIOMotorDataba
 
     await check_mongo_version(db_client)
 
+    if not skip_revision_check:
+        await check_revision(db_client[db_name])
+
     return db_client[db_name]
 
 
+async def check_revision(db: AsyncIOMotorDatabase):
+    """
+    Check if the required MongoDB revision has been applied.
+
+    Log a fatal error and exit if the required revision
+    has not been applied.
+
+    :param db: the application database object
+    """
+    if not await db.migrations.find_one({"revision_id": REQUIRED_MONGODB_REVISION}):
+        logger.fatal("Virtool does not have the required MongoDB revision.")
```

Comment:
Not sure if the string interpolation I have provided is correct.
```suggestion
        logger.fatal("The required MongoDB revision has not been applied: %s.", REQUIRED_MONGODB_REVISION)
```